python/tests/lang/lsp/base_lsp_test_case.py

```diff
@@ -118,51 +117,127 @@ def to_uri(self, file_name: str) -> Optional[str]:
-            )
-        )
-        return HoverResponse(response.result())
-
-    async def complete(self, file_name: str, line: int = 0, character: int = 0, trigger_kind: CompletionTriggerKind = CompletionTriggerKind.TriggerCharacter, trigger_character: str = SPACE_TRIGGER) -> CompletionResponse:
-        """
-        Send a code completion request and return the response.
-
-        Args:
-            file_name (str): The name of the virtual document in which to perform the code completion action.
-            line (int): The line number (starting from 0) at which to perform the code completion action.
-            character (int): The character number (starting from 0) at which to perform the code completion action.
-            trigger_kind (CompletionTriggerKind): The action that triggered the code completion action.
-            trigger_caracter (str): The character that triggered/triggers the code completion action.
-
-        Returns:
-            A CompletionResponse that is returned from the LSP server.
-        """
-        assert self.documents.get(file_name), f"Could not execute code completion in virtual document because there is no document named {file_name}."
-
-        response = await self.client.send_request(
-            methods.COMPLETION,
-            CompletionParams(
-                context=CompletionContext(trigger_kind=trigger_kind, trigger_character=trigger_character),
-                text_document=TextDocumentIdentifier(uri=self.to_uri(file_name)),
-                position=Position(line=line, character=character),
-            )
+    def virtual_document_to_lsp_document(self, file_name: str) -> Document:
+        """Convert a virtual document to an LSP document."""
+        assert self.documents.get(file_name), f"Could not convert virtual document because there is no document named {file_name}."
+        document = self.documents.get(file_name)
+        return Document(uri=document.file_name, source=document.content, version=document.version, sync_kind=TextDocumentSyncKind.FULL)
+
+    def build_text_document_position_params(self, file_name: str, line: int = 0, character: int = 0) -> dict:
+        """Return a dictionary that can be used as TextDocumentPositionParams in an LSP request."""
+        return {
+            "text_document": TextDocumentIdentifier(uri=self.to_uri(file_name)),
+            "position": Position(line=line, character=character),
+        }
+
+    async def build_request(self, file_name: str, response_type: type, method: str, params: Model):
+        self.assertIsNotNone(
+            self.documents.get(file_name),
+            f"Could not execute {method} in virtual document because there is no document named {file_name}."
         )
-        return CompletionResponse(response.result())
+        response = await self.client.send_request(method, params)
+        return response_type(response.result())
+
+
+TEST_DOCUMENT_NAME = "test.aac"
```

Comment:
If we're going to consolidate constants for testing, let's move it into a constants file in the `tests/lang/lsp` or `test.helpers` package.